### PR TITLE
docs: improve docstrings for BooleanValue and BooleanColumn

### PR DESCRIPTION
**Pull request by anjakefala**
Examples for `notany` and `notall` are incomplete because of this: https://github.com/ibis-project/ibis/issues/5988

### CODE REVIEW

File: ibis/expr/types/logical.py [hunk 1]
```diff
@@ -48,21 +48,184 @@ def ifelse(
...
+        -------
+        BooleanValue
+            A Boolean expression
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [[1], [], [42, 42], None]})
+        >>> t.arr.contains(42).__and__(t.arr.contains(1))
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:71]:
```suggestion
        >>> t.arr.contains(42) & t.arr.contains(1)
```

It's very uncommon for users to call "dunder" methods directly.

File: ibis/expr/types/logical.py [hunk 2]
```diff
@@ -48,21 +48,184 @@ def ifelse(
...
+        ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
+        │ boolean                                            │
+        ├────────────────────────────────────────────────────┤
+        │ False                                              │
+        │ False                                              │
+        │ False                                              │
+        │ NULL                                               │
+        └────────────────────────────────────────────────────┘
+
+        >>> t.arr.contains(42).__and__(t.arr.contains(42))
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:83]:
```suggestion
        >>> t.arr.contains(42) & t.arr.contains(42)
```

File: ibis/expr/types/logical.py [hunk 3]
```diff
@@ -48,21 +48,184 @@ def ifelse(
...
+        -------
+        BooleanValue
+            A Boolean expression
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [1, 2, 3, None]})
+        >>> (t.arr > 1).__or__(t.arr > 2)
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:120]:
```suggestion
        >>> (t.arr > 1) | (t.arr > 2)
```

File: ibis/expr/types/logical.py [hunk 4]
```diff
@@ -48,21 +48,184 @@ def ifelse(
...
+        ┡━━━━━━━━━━━━━━━━━┩
+        │ boolean         │
+        ├─────────────────┤
+        │ False           │
+        │ False           │
+        │ True            │
+        │ NULL            │
+        └─────────────────┘
+
+        >>> (t.arr == 2).__xor__(t.arr > 2)
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:181]:
```suggestion
        >>> (t.arr == 2) ^ (t.arr > 2)
```

File: ibis/expr/types/logical.py [hunk 5]
```diff
@@ -48,21 +48,184 @@ def ifelse(
...
+        -------
+        BooleanValue
+            A Boolean expression
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [1, 2, 3, None]})
+        >>> (t.arr == 2)
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:157]:
```suggestion
        >>> t.arr == 2
```

File: ibis/expr/types/logical.py [hunk 6]
```diff
@@ -48,21 +48,184 @@ def ifelse(
...
+        ┡━━━━━━━━━━━━━━━━┩
+        │ boolean        │
+        ├────────────────┤
+        │ False          │
+        │ True           │
+        │ False          │
+        │ NULL           │
+        └────────────────┘
+
+        >>> (t.arr > 2)
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:169]:
```suggestion
        >>> t.arr > 2
```

File: ibis/expr/types/logical.py [hunk 7]
```diff
@@ -48,21 +48,184 @@ def ifelse(
...
+        -------
+        BooleanValue
+            A Boolean expression
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [True, False, False, None]})
+        >>> t.arr.__invert__()
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:217]:
```suggestion
        >>> ~t.arr
```

File: ibis/expr/types/logical.py [hunk 8]
```diff
@@ -84,29 +247,178 @@ def any(self, where: BooleanValue | None = None) -> BooleanValue:
...
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [1, 2, 3, None]})
+        >>> (t.arr > 2).any()
+        True
+        >>> (t.arr > 4).any()
+        False
+        >>> m = ibis.memtable({"arr": [True, True, True, False]})
+        >>> (t.arr == None).any(where=(t.arr != None))
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:265]:
```suggestion
        >>> (t.arr == None).any(where=t.arr != None)
```

File: ibis/expr/types/logical.py [hunk 9]
```diff
@@ -84,29 +247,178 @@ def any(self, where: BooleanValue | None = None) -> BooleanValue:
...
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [1, 2, 3, 4]})
+        >>> (t.arr >= 1).all()
+        True
+        >>> (t.arr > 2).all()
+        False
+        >>> (t.arr == 2).all(where=(t.arr == 2))
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:318]:
```suggestion
        >>> (t.arr == 2).all(where=t.arr == 2)
```

File: ibis/expr/types/logical.py [hunk 10]
```diff
@@ -84,29 +247,178 @@ def any(self, where: BooleanValue | None = None) -> BooleanValue:
...
+        True
+        >>> (t.arr == 2).all(where=(t.arr >= 2))
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:320]:
```suggestion
        >>> (t.arr == 2).all(where=t.arr >= 2)
```

File: ibis/expr/types/logical.py [hunk 11]
```diff
@@ -84,29 +247,178 @@ def any(self, where: BooleanValue | None = None) -> BooleanValue:
...
+        Returns
+        -------
+        BooleanColumns
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [1, 2, 3, 4]})
+        >>> ((t.arr > 1).__or__(t.arr >= 1)).cumany()
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:362]:
```suggestion
        >>> ((t.arr > 1) | (t.arr >= 1)).cumany()
```

File: ibis/expr/types/logical.py [hunk 12]
```diff
@@ -84,29 +247,178 @@ def any(self, where: BooleanValue | None = None) -> BooleanValue:
...
+        ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
+        │ boolean                                                  │
+        ├──────────────────────────────────────────────────────────┤
+        │ True                                                     │
+        │ True                                                     │
+        │ True                                                     │
+        │ True                                                     │
+        └──────────────────────────────────────────────────────────┘
+
+        >>> ((t.arr > 1).__and__(t.arr >= 1)).cumany()
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:374]:
```suggestion
        >>> ((t.arr > 1) & (t.arr >= 1)).cumany()
```

File: ibis/expr/types/logical.py [hunk 13]
```diff
@@ -84,29 +247,178 @@ def any(self, where: BooleanValue | None = None) -> BooleanValue:
...
+        Returns
+        -------
+        BooleanColumns
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.memtable({"arr": [1, 2, 3, 4]})
+        >>> ((t.arr > 1).__and__(t.arr >= 1)).cumall()
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:400]:
```suggestion
        >>> ((t.arr > 1) & (t.arr >= 1)).cumall()
```

File: ibis/expr/types/logical.py [hunk 14]
```diff
@@ -84,29 +247,178 @@ def any(self, where: BooleanValue | None = None) -> BooleanValue:
...
+        ┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
+        │ boolean                                                   │
+        ├───────────────────────────────────────────────────────────┤
+        │ False                                                     │
+        │ False                                                     │
+        │ False                                                     │
+        │ False                                                     │
+        └───────────────────────────────────────────────────────────┘
+
+        >>> ((t.arr > 0).__and__(t.arr >= 1)).cumall()
```
Comment (cpcloud) [inline ibis/expr/types/logical.py:412]:
```suggestion
        >>> ((t.arr > 0) & (t.arr >= 1)).cumall()
```

**Review by cpcloud — CHANGES_REQUESTED** [review]
Some style nits, but otherwise LGTM!

**Comment by cpcloud** [comment]
@anjakefala Yup, sounds like a plan!

File: ibis/expr/types/logical.py (hunk 7 above)
Comment (cpcloud) [reply ibis/expr/types/logical.py:217]: @anjakefala Looks like maybe this didn't get picked up in the changes you pushed?

Comment (anjakefala) [reply ibis/expr/types/logical.py:217]: Good eye!

**Pull request merged**